src/theme/useFont.js

```diff
@@ -0,0 +1,50 @@
+import { useState, useLayoutEffect } from "react";
+import lightTheme from "./lightTheme";
+
+const useFont = (customFontConfig) => {
+  const [fontConfig, setFontConfig] = useState(null);
+  useLayoutEffect(() => {
+    if (customFontConfig) {
+      setFontConfig(customFontConfig);
+      applyFonyAssets(customFontConfig);
+    }
+  }, [customFontConfig]);
+  return [fontConfig, setFontConfig];
+};
+
+const applyFonyAssets = (fontConfig) => {
+  var newStyle = document.createElement("style");
+  newStyle.appendChild(
+    document.createTextNode(`
+@font-face {
+  font-family: ${fontConfig.fontFamilyText};
+  src: url(${fontConfig.semiBoldUrl})
+    format("${fontConfig.format}");
+  font-weight: ${lightTheme["font-weight-semi-bold"]};
```

Comment:
We have small theme dependency that may cause inconsistent behavior when consumer overrides the default `font-weight-semi-bold, font-weight-light, font-weight-regular` values.

But this should be resolved later when we change the `useTheme` API to be `const [theme, useTheme] = useTheme(...)` which will give us the possibility to call `useTheme` hook to get the weight values instead of importing default theme.